### DISCUSSION TITLE
Requesting for Feedback on Specific Unit Names for Building Energy Metrics

### DISCUSSION THREAD

**Discussion by connorjcantrell** [qudt/qudt-public-repo#725] [General]
Hi @steveraysteveray,

Mapped is in the process of extending our ontology to capture Building Energy Metrics, specifically those used by EnergyStar for tracking Greenhouse Gas (GHG) emissions. We would like to ensure that these metrics and the units of measurement are in line with the QUDT ontology.

We would appreciate your feedback on the following units we are utilizing for different metrics:
- Metric Tons CO2e: This unit is used to measure the quantity of GHG Emissions, specifically for National and Regional Median Total GHG Emissions.
- kgCO2e/ft²: This unit is used to calculate the intensity of Total Location-Based GHG Emissions.
- kgCO2e/MBtu: This unit is used for evaluating the "Emissions Factor", particularly for Location-Based Electricity/eGRID emissions.
- GJ/m²: This is the metric equivalent of the previous unit and is utilized in similar contexts for measurements in metric systems.
- kgCO2e/m²: Used to measure the intensity of GHG emissions in metric units.
- kgCO2e/GJ: This unit is used for calculating the GHG Emissions Factor, particularly for Direct Fuels, District Steam, District Hot Water, and District Chilled Water in metric systems.

More in particular, I am seeking feedback on how we could introduce units related to CO2e, which a universal unit for measure Greenhouse Gases. Please see the following [definition by EnergyStar](https://portfoliomanager.energystar.gov/pm/glossary#GHGemissions):
> GHG Emissions
Greenhouse Gas (GHG) Emissions are the carbon dioxide (CO2), methane (CH4), and nitrous oxide (N2O) gases released into the atmosphere as a result of energy consumption at the property. GHG emissions are expressed in carbon dioxide equivalent (CO2e), a universal unit of measure that combines the quantity and global warming potential of each greenhouse gas. Emissions are reported in four categories, each is available as a total amount in metric tons (Metric Tons CO2e) or as an intensity value in kilograms per square foot (kgCO2e/ft2)

At this time, we are seeking to reach agreement on a name for the units listed above. I can begin implementing these units over the next couple of weeks and create a PR, assuming the QUDT is open to introducing these units to its ontology.

Thank you!
Connor

**Comment by steveraysteveray** [comment]
We would welcome these units! Please let us know if you have any questions about the localnames. We have attempted to formulate a comprehensive grammar for these in our [wiki](https://github.com/qudt/qudt-public-repo/wiki/Unit-Vocabulary-Submission-Guidelines). Regarding CO2e, my suggestion is to treat that as a qualifier of KiloGM, as in KiloGM_CO2E. (Note that our unit localnames are all uppercase except for the prefixes). This is similar to how we name units like unit:IN_H2O for "inches of water" as a measure of pressure. The quantity kind for unit:IN_H2O is quantitykind:ForcePerArea (i.e. pressure), even though one might naively think it is a measure of length. In your case, the quantity kind for  unit:KiloGM_CO2E would presumably be quantitykind:Mass even though it is being used as a proxy for emissions, right? Or is it more like IN_H2O? I admit I'm a bit confused when you say GJ/m² is the metric equivalent of kgCO2e/MBtu, because the first is energy/area and the second is mass(equivalent)/energy.

Looking forward to more discussion and hopefully a PR.

**Comment by connorjcantrell** [comment]
I'd like to make a correction, GJ/m² is the unit used to determine the energy a property consumes divided by the property's square feet/ meter. I just learned that they also use kBTU/ft², which is already defined.

> the quantity kind for unit:KiloGM_CO2E would presumably be quantitykind:Mass
Yes, I agree that it would be mass

In addition to defining the quantity kind of `KiloGM_CO2E`, I will implement the following units:
- `KiloGM_CO2E-PER-FT2`
- `KiloGM_CO2E-PER-MegaBTU_IT`
- `GigaJ-PER-M2`
- `KiloGM_CO2E-PER-M2`
- `KiloGM_CO2E-PER-GigaJ`

A building's GHG emissions (aka CO2e) is measured in "Metric Tons". Would I need to define a unit of `TON_Metric_CO2E`? Not entirely sure on this one.

**Comment by steveraysteveray** [comment]
You don't need to define a quantity kind of KiloGM_CO2E, just the unit KiloGM_CO2E. The quantity kind is quantitykind:Mass

On your second question, about Metric Tons, that really depends on what your community expects. Just using TON_Metric is fine, with the assumption that the context makes it clear what you are talking about.

(I'm also embarrassed to realize that we never changed TON_Metric to TON_METRIC. We don't like to break people's applications, so we would create the new one and leave the old one, marked qudt:deprecated True.   )

**Reply by connorjcantrell** [reply]
Ok, I like that idea. We will just use TON_METRIC, rather than define a new unit. Ok great, establishing the names is what I was hoping for today, so thanks for getting back to me.

I will plan on creating a PR next week.

**Comment by steveraysteveray** [comment]
One quick update. The QUDT Board just discussed the issue of uppercase/Camelcase for qualifiers (like _METRIC vs. _Metric) and we have agreed that unit URIs are all uppercase EXCEPT for prefixes and for qualifiers. So the implications for you are:

KiloGM_CO2e
and
TON_Metric

...as the respective URIs. I will be clarifying our wiki guidance shortly.

Thanks again.

**Reply by connorjcantrell** [reply]
Thank you for the update!

**Comment by connorjcantrell** [comment]
Hey one more quick update. We have decided against using the target substance in the unit name (i.e. CO2e).

- Metric Tons CO2e -> Metric Tons
- kgCO2e/ft² -> kg/ft²
- kgCO2e/MBtu -> kg/MBtu
- kgCO2e/m² -> kg/m²
- kgCO2e/GJ -> kg/GJ

It is possible that these units already exist in your ontology. I will create a PR if any additions are needed over the next several days

**Comment by steveraysteveray** [comment]
That makes sense also. I wasn't sure what your terms of art are. I know that some of your desired units will need to be added.

**Comment by connorjcantrell** [comment]
@steveraysteveray I might need to reconsider not including CO2e in our proposed unit names. An [article](https://www.techtarget.com/Sustainability/feature/CO2-vs-CO2e-What-is-the-difference-and-why-does-it-matter#:~:text=CO2e%20uses%20CO2%20as%20a,produce%20into%20a%20single%20number) I came across clarified that CO2e is essential for measuring GHG emissions, using CO2 as the standard for comparison.

Take, for instance, having 1 kg each of carbon dioxide, methane, and nitrous oxide. While it might seem like it totals to 3 kg, in reality, it equates to 324 kg/CO2e when considering their respective global warming potentials.

If you are not opposed, I would like to create a PR for the addition of all previously proposed CO2e relating units.

Thank you

**Reply by jbkoh** [reply]
To be clear, we stepped back last time because thought e in CO2e stands for emission, but it's actually "equivalent".

**Comment by jhodgesatmb** [comment]
CO2e doesn't seem to comply with QUDT naming conventions. Whatever 'e' is
intended to mean, it seems to be a qualifier and so should be CO2_e. Even
then I suspect that we would say CO2_E or even something else. Having not
followed any of this discussion I may be off base but it seemed as though
this needed to be articulated.

On Mon, Nov 20, 2023 at 3:13 PM Jason B. Koh ***@***.***>
wrote:

> To be clear, we stepped back last time because thought e in CO2e stands
> for emission, but it's actually "equivalent".
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/qudt/qudt-public-repo/discussions/725#discussioncomment-7624451>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATQRWJOGS2ZML4BXGRMQZDYFPP25AVCNFSM6AAAAAA2QI4ECGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMMRUGQ2TC>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Jack

**Comment by connorjcantrell** [comment]
@steveraysteveray @jhodgesatmb 

CO2 equivalent is a non SI unit defined by the Intergovernmental Panel on Climate Change (IPCC). Please see [Metrics and Methodology](https://www.ipcc.ch/site/assets/uploads/2018/02/ipcc_wg3_ar5_annex-ii.pdf), which was issued by the IPCC. You will find CO2eq defined in section "A.II.1 Standard units and unit conversion" under table A.II.3.

### Use cases:
[RTE](https://en.wikipedia.org/wiki/R%C3%A9seau_de_Transport_d'%C3%89lectricit%C3%A9) France uses the following units:
- [gCO2eq/kWh](https://www.rte-france.com/en/home)
- tCO2eq/MWh

[EnergyStar](https://portfoliomanager.energystar.gov/pdf/reference/Emissions.pdf) uses the following units:
- kgCO2eq/MBtu
- gCO2eq/L
- kgCO2eq/tonne

<img width="690" alt="Screenshot 2023-11-28 at 9 23 01 AM" src="https://github.com/qudt/qudt-public-repo/assets/33592545/53189f01-54d0-4cb2-b8dd-feb1abfb9e05">



In addition to the above units, I would also like to define units [previously discussed](https://github.com/qudt/qudt-public-repo/discussions/725#discussioncomment-6493247) above. Please let me know if you need additional references. Looking forward to receiving your feedback.

Final thought, it may be worth considering using  `CO2eq`, `CO2_eq`, or `CO2_EQ`, rather than CO2e or CO2E.
